modules/identity/keys.go

```diff
@@ -37,17 +44,36 @@ func GenerateKeyPair() (k KeyPair, err error) {
 func (k *KeyPair) Save(path string) error {
 	seed := k.PrivateKey.Seed()
 
-	return ioutil.WriteFile(path, seed, 0400)
+	return versioned.WriteFile(path, seedVersionLatest, seed, 0400)
 }
 
-// LoadSeed reads a seed from a file located at path and re-create a
+// LoadSeed from path
+func LoadSeed(path string) ([]byte, error) {
+	version, seed, err := versioned.ReadFile(path)
+	if versioned.IsNotVersioned(err) {
+		// this is a compatibility code for seed files
+		// in case it does not have any version information
+		versioned.WriteFile(path, seedVersionLatest, seed, 0400)
+		version = seedVersionLatest
+	} else if err != nil {
+		return nil, err
+	}
+
+	if version.NE(seedVersionLatest) {
+		return nil, fmt.Errorf("unknown seed version")
+	}
+
+	return seed, nil
+}
+
+// LoadKeyPair reads a seed from a file located at path and re-create a
 // KeyPair using the seed
-func LoadSeed(path string) (k KeyPair, err error) {
-	seed, err := ioutil.ReadFile(path)
+func LoadKeyPair(path string) (k KeyPair, err error) {
+	seed, err := LoadSeed(path)
 	if err != nil {
 		return k, err
 	}
-
+	// version update can go here.
```

Comment:
How do we know the version of the seed we just loaded at this point ? I don't see how we can decide to do an ugprade or not here.

Reply:
Ah, I moved the code but didn't move the comment :/